packages/react-devtools-shared/src/devtools/ContextMenu/useContextMenu.js

```diff
@@ -0,0 +1,32 @@
+import {useContext, useEffect} from 'react';
+import {RegistryContext} from './Contexts';
+
+export default function useContextMenu({data, id, ref}) {
+  const {showMenu} = useContext(RegistryContext);
+
+  useEffect(
+    () => {
+      if (ref.current !== null) {
+        const handleContextMenu = event => {
+          event.preventDefault();
+          event.stopPropagation();
+
+          const pageX =
+            event.clientX || (event.touches && event.touches[0].pageX);
+          const pageY =
+            event.clientY || (event.touches && event.touches[0].pageY);
```

Comment:
This confused me a bit. I was surprised the property is called `pageX/Y` everywhere else (e.g. in `showMenu` API and things it calls) because intuitively I expected `clientX/Y` to be used.

But it seems like we *do* use `clientX/Y` first.

What is the case where it doesn't exist? And does `pageX/Y` give us the same thing in that fallback? What if we already scrolled a bit, aren't they different in their meaning?

Reply:
TBH this is based on my looking at other context menu implementations. 😄 It seems to work from my testing.

Reply:
I'm confused because `client*` and `page*` are two different coordinate systems. If a page is scrolled down, `page*` will include how much you scrolled, but `client*` wouldn't.

If we use this value to calculate where to position the menu, it doesn't make sense to me that `page*` would be accurate. What matters is how far it is in the viewport — not relative to the *page*. So I would expect to see a bug there when this branch gets hit. When you say it works in testing, do you mean you specifically tested when

- the view is scrolled down, and
- we're using touch events

?

Because that's the case where I'd expect a bug.

My suggested change would be to always use `client*`, e.g. `event.clientX || (event.touches && event.touches[0].clientX)`. If we don't care to test the "touch" code path, it seems to me that we should start with what makes most sense (`client*` represents client coordinates), and only use `page*` if we have a good reason for that (my understanding is we don't have one?).

In either case, if normally we use `client*` (because most people using DevTools don't use touch), it would make sense to me to *rename the properties and arguments* to be `client*` instead of `page*`. They don't have effect on semantics but they are confusing when they don't match what you really get. E.g. I was looking at the measurement implementation and I was confident there was a bug there due to `page*` usage, but only later I found we're actually passing `e.client*` there.

Reply:
> Googling these exact lines didn't give me results.

Presumably because Prettier formatted these particular lines in a different way.

After thinking and testing a little more, I believe:
1. It probably doesn't matter which one we use, since the main DevTools window never scrolls, _but_
2. If it did scroll, we'd want to use `pageY` in both places (not just in the touch event) because otherwise our context menu would potentially be positioned too above where the cursor was.